### PR TITLE
status for more core types

### PR DESCRIPTION
**Pull request by barney-s**
#92

### CODE REVIEW

File: pkg/apis/app/v1beta1/status.go [hunk 1]
```diff
@@ -82,6 +99,94 @@ func stsStatus(rsrc *appsv1.StatefulSet) string {
...
+				progress = false
+			}
+		case appsv1.DeploymentAvailable:
+			if c.Status == corev1.ConditionFalse {
+				available = false
+			}
+		}
+	}
+
+	if progress && available {
```
Comment (huyhg) [inline pkg/apis/app/v1beta1/status.go]: Can you help explain a bit more about the `progress` bool and the for loop above? Shouldn't this condition be `!progress && available`?

File: pkg/apis/app/v1beta1/status.go [hunk 2]
```diff
@@ -82,6 +99,94 @@ func stsStatus(rsrc *appsv1.StatefulSet) string {
...
+// Replicaset
+func replicasetStatus(rsrc *appsv1.ReplicaSet) string {
+	status := StatusInProgress
+	failure := false
+	for _, c := range rsrc.Status.Conditions {
+		switch c.Type {
+		// https://github.com/kubernetes/kubernetes/blob/a3ccea9d8743f2ff82e41b6c2af6dc2c41dc7b10/pkg/controller/replicaset/replica_set_utils.go
+		case appsv1.ReplicaSetReplicaFailure:
+			if c.Status == corev1.ConditionTrue {
+				failure = true
```
[the remark this answers is not in the document]
Comment (barney-s) [reply pkg/apis/app/v1beta1/status.go]: ack

File: pkg/apis/app/v1beta1/status.go [hunk 3]
```diff
@@ -82,6 +99,94 @@ func stsStatus(rsrc *appsv1.StatefulSet) string {
...
+func serviceStatus(rsrc *corev1.Service) string {
+	status := StatusReady
+	return status
+}
+
+// Pod
+func podStatus(rsrc *corev1.Pod) string {
+	status := StatusInProgress
+	for i := range rsrc.Status.Conditions {
+		if rsrc.Status.Conditions[i].Type != corev1.PodReady {
```
[the remark this answers is not in the document]
Comment (barney-s) [reply pkg/apis/app/v1beta1/status.go]: ack

File: pkg/apis/app/v1beta1/status.go (hunk 1 above)
Comment (barney-s) [reply pkg/apis/app/v1beta1/status.go]: Progressing is a continuous status here.
If it is not progressing it is due to an error.
It is not an indication of upgrade in progress.

https://github.com/kubernetes/kubernetes/blob/a3ccea9d8743f2ff82e41b6c2af6dc2c41dc7b10/pkg/controller/deployment/progress.go#L85

**Pull request merged**